include/kodlab_mjbots_sdk/behavior_manager.h

```diff
@@ -0,0 +1,295 @@
+ *          and transitions between them.
+ * @note By default, this class initializes its internal behaviors list with a
+ *       `kodlab::Offbehavior`. This is done for safety purposes so that, when
+ *       turned on, the robot does not command some undefined torques. This
+ *       default can be overwritten with the `SetDefaultBehavior` member.
+ * @tparam Robot[optional] derived `kodlab::RobotBase` class
+ */
+template<class Robot = kodlab::RobotBase>
+class BehaviorManager
+{
+  static_assert(std::is_base_of<kodlab::RobotBase, Robot>::value,
+                "Robot must have base kodlab::RobotBase.");
+
+private:
+  /**
+   * @brief Robot that behaviors run on
+   */
+  std::shared_ptr<Robot> robot_;
+
+  /**
+   * @brief Behavior command that kills robot, default -1
+   */
+  static const int KILL_ROBOT_IDX = -1;
+
+  /**
+   * @brief Behavior command that turns robot off, always 0
+   */
+  static const int DEFAULT_IDX = 0;
+
+  /**
+   * @brief Behaviors available to the robot
+   * @note The first element of this vector is always `OFF_BEHAVIOR`.  Added
+   *       behaviors are appended to the end.
+   */
+  std::vector<std::unique_ptr<Behavior<Robot>>> behaviors_;
+
+  /**
+   * @brief Names of behaviors in `behaviors_`
+   */
+  std::vector<std::string> names_;
+
+  /**
+   * @brief Index in `behaviors_` of the selected behavior
+   */
+  unsigned int selected_idx = 0;
+
+  /**
+   * @brief Index in `behaviors_` of the incoming behavior while transitioning
+   *        behaviors
+   */
+  int next_idx_ = 0;
+
+  /**
+   * @brief Indicates whether a behavior switch is in progress
+   */
+  bool switching_behaviors_ = false;
+
+public:
+  /**
+   * @brief Construct a behavior manager object
+   * @param robot pointer to robot that the behavior is executing on
+   */
+  BehaviorManager(std::shared_ptr<Robot> robot) : robot_(robot)
+  {
+    AddBehavior<OffBehavior<Robot>>(robot_, "OFF");
+  }
+
+  /**
+   * @brief Destructor
+   */
+  virtual ~BehaviorManager() {}
+
+  /**
+   * @brief Set the current behavior
+   * @param next_idx index in `behaviors_`
+   */
+  virtual void SetBehavior(const int &next_idx)
+  {
+    if (next_idx == KILL_ROBOT_IDX)
+    {
+      kodlab::CTRL_C_DETECTED = true; // kill robot
+      LOG_FATAL("Behavior %d received, killing robot.", KILL_ROBOT_IDX);
+    } else if (next_idx == selected_idx)
+    {
+      LOG_INFO("Behavior %d is already active.", next_idx);
+      return;
+    }
+    else if (next_idx < behaviors_.size() && next_idx >= 0)
+    {
+      next_idx_ = next_idx; // update next behavior
+      if (!behaviors_[next_idx_]->is_initialized())
+      {
+        behaviors_[next_idx_]->Init();
+      }
+      behaviors_[selected_idx]->Stop(*behaviors_[next_idx_]); // stop current behavior
+      switching_behaviors_ = true;  // indicate transition is in progress
+    } else
+    {
+      LOG_WARN("Invalid behavior selected: %d", next_idx);
```

Comment:
I feel like we should not just warn when this happens.

Reply:
For behaviors, I think that warning is sufficient.  The way that @jdcaporale and I use this involves a simple CLI which prompts for a mode number and publishes it.  It is not uncommon for me to incorrectly enter a mode that is out of range.  In this case, I would prefer just to be warned than have the robot shut off or send the motors into braking.

If you feel differently, perhaps we could add an optional parameter when constructing the `BehaviorManager` to dictate what the robot should do in this circumstance (i.e., warn, return to default behavior, or kill).